code/modules/components/ai/ai_component.dm

```diff
@@ -0,0 +1,13 @@
+/datum/component/ai
+	var/datum/component/controller/controller
+
+	var/state=0 // AI_STATE_* of the AI.
+
+/datum/component/ai/RecieveSignal(var/message_type, var/list/args)
+	switch(message_type)
+		if("state") // list("name"="statename")
+			state = args["name"]
+
+/datum/component/ai/New(var/datum/component_container/CC)
+	..(CC)
```

Comment:
I like passing arguments explicitly to confuse myself a bit less.  If it's a code quality requirement, I can do it.